buildrunner/config/models_step.py

```diff
@@ -102,49 +132,92 @@ class StepRun(RunAndServicesBase, extra="forbid"):
     ssh_keys: Optional[List[str]] = Field(alias="ssh-keys", default=None)
     artifacts: Optional[Dict[str, Optional[AnnotatedArtifact]]] = None
     platform: Optional[str] = None
-    cap_add: Optional[Union[str, List[str]]] = None
+    cap_add: Optional[List[str]] = None
     privileged: Optional[bool] = None
-    post_build: Optional[Union[str, Dict[str, Any]]] = Field(
-        alias="post-build", default=None
-    )
+    post_build: Optional[StepBuild] = Field(alias="post-build", default=None)
     no_cache: Optional[bool] = Field(alias="no-cache", default=None)
 
+    @field_validator("post_build", mode="before")
```

Comment:
I'm open to other names here, I was trying to just say that it transforms the data for post_build to be consistent always (i.e. a dict instead a str/list/whatever) for each of these methods. The name may be confusing here since the field is `post_build`? I'm not sure

Reply:
I got it after reading but it was confusing at first, it's definitely because of the field name that it's a little confusing. It's fine. I just worry that someone like me will come and say, 'this dummy made a post function pre validate' and change it 😓